### ISSUE TITLE
build failure with GCC11

### ISSUE THREAD

**Issue by Alessandro-Barbieri** [bsc-pm/nanos6#2]
Log here: https://786717.bugs.gentoo.org/attachment.cgi?id=703788
```
libtool: compile:  x86_64-pc-linux-gnu-g++ -DHAVE_CONFIG_H -I. -I/usr/include -DBOOST_ENABLE_ASSERT_DEBUG_HANDLER -pthread -I/usr/include -I./src/executors/workflow/cluster/null -I./src/cluster/null -I./src/cluster/offloading/null -I./src/hardware/cluster/null -I./src/hardware/cluster/memory/null -I./src/executors/workflow/null -I./src/support/chronometers/std -I./src/executors/threads/cpu-managers/default -I./src/memory/allocator/malloc -I./src/memory/vmm/smp -I./src/instrument/null -Ino/include -Wall -Wextra -Wdisabled-optimization -Wshadow -fvisibility=hidden -O0 -g3 -fstack-protector-all -I./src -I./src -I./api -I. -faligned-new -I./src/executors/threads/kernel-level -I./src/instrument/support/introspection/ElfUtils -I./src/instrument/support/backtrace/backtrace -I./src/dependencies/linear-regions -I./src/dependencies/linear-regions-fragmented -Wall -Wextra -Wdisabled-optimization -Wshadow -fvisibility=hidden -c src/hardware-counters/rapl/RAPLHardwareCounters.cpp  -fPIC -DPIC -o src/hardware-counters/rapl/.libs/libnanos6_debug_regions_la-RAPLHardwareCounters.o
In file included from src/hardware/places/MemoryPlace.hpp:12,
                 from src/hardware/places/NUMAPlace.hpp:10,
                 from src/hardware/places/NUMAPlace.cpp:7:
./src/memory/AddressSpace.hpp:27:9: error: ‘size_t’ does not name a type
   27 |         size_t getMemoryPlacesCount() const
      |         ^~~~~~
./src/memory/AddressSpace.hpp:1:1: note: ‘size_t’ is defined in header ‘<cstddef>’; did you forget to ‘#include <cstddef>’?
  +++ |+#include <cstddef>
    1 | /*
```
https://bugs.gentoo.org/786717

**Comment by kevinsala** [comment]
Hi Alessandro,
Tomorrow we will publish a new release fixing that error and adding other features. As a temporary workaround, you can add the following include in the file `src/memory/AddressSpace.hpp`:

```
#include <cstddef>
```
That should make the compilation error to disappear. Thank you for reporting the issue.

**Comment by kevinsala** [comment]
This error should already be fixed in the Nanos6 version 2.6 (OmpSs-2 2021.06) that we have just published. To use that new version you will need to update the Mercurium or Clang compiler to their new version, as specified in [OmpSs-2 Releases](https://github.com/bsc-pm/ompss-2-releases).
